### PR TITLE
feat: pluggy

### PR DESCRIPTION
**Pull request by bram2000**

### CODE REVIEW

File: digital_land/cli.py [hunk 1]
```diff
@@ -310,7 +316,7 @@ def pipeline_cmd(input_path, output_path, null_path, issue_path, use_patch_callb
         collection,
         Organisation().organisation_uri,
         patch,
-        use_patch_callback,
+        pm,
     )
```
Comment (ksp37) [inline digital_land/cli.py:320]: No `pm.register(harmoniser)` needed? (Like L255)

File: digital_land/harmonise.py [hunk 2]
```diff
@@ -153,12 +152,13 @@ def harmonise(self, reader):
...
-                if self.harmonise_callback and hasattr(
-                    self.harmonise_callback, "patch"
-                ):
-                    row[field] = self.harmonise_callback.patch(
-                        field, row[field], self.log_issue
+                if self.plugin_manager:
+                    plugin_results = self.plugin_manager.hook.apply_patch_post(
+                        fieldname=field, value=row[field]
                     )
+                    if len(plugin_results) == 1:
```
Comment (ksp37) [inline digital_land/harmonise.py:159]: It can return an array?

Comment (bram2000) [reply digital_land/harmonise.py:159]: yes, `plugin_manager.hook.<function>` will always return a list, as you might have multiple implementations for a given hook.

File: digital_land/cli.py (hunk 1 above)
Comment (bram2000) [reply digital_land/cli.py:320]: good spot, this may actually be better in `__init__` of harmoniser, so it always happens. Will change now.

**Review by ksp37 — APPROVED** [review]

**Pull request merged**